oshmem/mca/atomic/atomic.h

```diff
@@ -86,17 +86,25 @@ struct mca_atomic_base_module_1_0_0_t {
     opal_object_t super;
 
     /* Collective function pointers */
+    int (*atomic_add)(void *target,
+                      const void *value,
```

Comment:
how about this optimization: instead of passing pointers (for "value" and "cond"), pass them as uint64_t number.
will avoid pointer dereference and some branches in the mca_atomic code

Reply:
it will not help because original shmem API operates by pointers, and we will have to convert ptr->value early using same if/else (size) construct

Reply:
but original shmem api uses values: for example https://www.open-mpi.org/doc/v3.0/man3/shmem_long_fadd.3.php - `long value`

Reply:
ok, done